workers/unity/Assets/Playground/Scripts/MonoBehaviours/FlashOnCollision.cs

```diff
@@ -0,0 +1,67 @@
+﻿using Generated.Playground;
+using Improbable.Gdk.Core.GameObjectRepresentation;
+using UnityEditor;
+using UnityEngine;
+using Color = UnityEngine.Color;
+
+public class FlashOnCollision : MonoBehaviour
+{
+    [Require] private Collisions.Reader reader;
+
+    private float collideTime;
+    private bool flashing = false;
+
+    [SerializeField] private float flashTime = 0.2f;
+
+    private MeshRenderer renderer;
+
+    private static MaterialPropertyBlock basicMaterial = new MaterialPropertyBlock();
+    private static MaterialPropertyBlock flashingMaterial = new MaterialPropertyBlock();
+
+    [RuntimeInitializeOnLoadMethod]
+    public static void SetupColors()
```

Comment:
This looks super overkill

Reply:
Its that or giving every single cube ever its own instance of a MaterialPropertyBlock

Reply:
OK, since the work is already done it's not worth changing, but why are we optimizing things so tightly at this phase though...